sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/MergeScalarSubqueriesSuite.scala

```diff
@@ -534,20 +534,10 @@ class MergeScalarSubqueriesSuite extends PlanTest {
 
   test("Merging subqueries from different places") {
     val subquery1 = ScalarSubquery(testRelation.select(('a + 1).as("a_plus1")))
-    val subquery2 = ScalarSubquery(testRelation.select(('a + 2).as("a_plus2")))
```

Comment:
Hi @dtenedor, thanks for reporting this bug and pinging me.

I think the test is a valid usecase so I'm thinking about if we could come up with a better fix. I think the root cause of the issue is in `tryMergePlans()`. When we try merging `Project`s or `Aggregate` and don't take into accout that `np` can contain `ScalarSubqueryReference`. So probably filtering out `ScalarSubqueryReference`s there should fix the bug but keep this usecase working. Also, I think `EXISTS_SUBQUERY`, `IN_SUBQUERY` should not matter.

Reply:
Thanks for the review!

I was able to make the fix you suggested instead, to fix the bug while preserving the functionality of this unit test case.

Reply:
🙏 thanks for your attention!

Reply:
Looks good, thanks for the fix!